File: 05-hash/Python/aluno.py
class Aluno:
    def __init__(self, matricula, nome):
        self.matricula = matricula
        self.nome = nome
        
    def __eq__(self, other):
        if isinstance(other, Aluno):
            return self.matricula == other.matricula
        return False
    
    def __gt__(self, other):
        if self.matricula == other.matricula:
            return self.nome > other.nome
        return self.matricula > other.matricula
    
    def __str__(self):
        return "Aluno{" + "matricula=" + str(self.matricula) + ", nome=" + self.nome + '}'
    
    def __hash__(self):    
        return hash((self.matricula))

File: 05-hash/Java/hash/src/main/java/com/mycompany/hash/Aluno.java

package com.mycompany.hash;

/**
 *
 * @author alexandrezamberlan
 */
public class Aluno implements Comparable <Aluno>{
    public int matricula;
    public String nome;

    public Aluno(int matricula, String nome) {
        this.matricula = matricula;
        this.nome = nome;
    }
    
    public Aluno(int matricula) {
        this.matricula = matricula;
    }
    
    @Override
    public int hashCode() {
        int hash = 7;
        hash = 41 * hash + this.matricula;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Aluno other = (Aluno) obj;
        if (this.matricula != other.matricula) {
            return false;
        }
        return true;
    }

    @Override
    public int compareTo(Aluno a) {
        if (this.matricula == a.matricula) {
            return 0;
        } else if (this.matricula < a.matricula) {
            return -1;
        }
        return 1;
    }

    @Override
    public String toString() {
        return "Aluno{" + "matricula=" + matricula + ", nome=" + nome + '}';
    }
}
